tests/test_controls.py

```diff
@@ -0,0 +1,526 @@
+"""Tests for control and procedure classes"""
+
+import pytest
+import pydantic
+from typing import Union, Any
+from RAT.controls import BaseProcedure, Calculate, Simplex, DE, NS, Dream, ControlsClass
+from RAT.utils.enums import ParallelOptions, Procedures, DisplayOptions, BoundHandlingOptions, StrategyOptions
+
+
+class TestBaseProcedure:
+    """
+    Tests the BaseProcedure class.
+    """
+
+    def setup_class(self):
+        self.base_procedure = BaseProcedure()
+
+    @pytest.mark.parametrize("property, value", [('parallel', ParallelOptions.Single),
+                                                 ('calcSldDuringFit', False),
+                                                 ('resamPars', [0.9, 50]),
+                                                 ('display', DisplayOptions.Iter)])
+    def test_base_property_values(self, property: str, value: Any) -> None:
```

Comment:
Add a docstring here giving a quick explanation of the behaviour you're expecting - have a look at the other tests.